netbox/dcim/models/device_components.py

```diff
@@ -547,10 +547,41 @@ class BaseInterface(models.Model):
         blank=True,
         verbose_name=_('bridge interface')
     )
+    untagged_vlan = models.ForeignKey(
+        to='ipam.VLAN',
+        on_delete=models.SET_NULL,
+        related_name='%(class)ss_as_untagged',
+        null=True,
+        blank=True,
+        verbose_name=_('untagged VLAN')
+    )
+    tagged_vlans = models.ManyToManyField(
+        to='ipam.VLAN',
+        related_name='%(class)ss_as_tagged',
+        blank=True,
+        verbose_name=_('tagged VLANs')
+    )
+    qinq_svlan = models.ForeignKey(
+        to='ipam.VLAN',
+        on_delete=models.SET_NULL,
+        related_name='%(class)ss_svlan',
+        null=True,
+        blank=True,
+        verbose_name=_('Q-inQ SVLAN')
+    )
 
     class Meta:
         abstract = True
 
+    def clean(self):
```

Comment:
If a VLAN is assigned to a Site should it restrict what Q-in-Q SVLAN it can be assigned to (i.e. in the same Site)?  Not sure if that makes sense or is overkill.

Reply:
It might be reasonable, but I want to wait until we get some feedback on the initial implementation before making that determination.